base/src/main/java/uk/ac/ebi/atlas/species/AtlasInformationDao.java

```diff
@@ -0,0 +1,44 @@
+package uk.ac.ebi.atlas.species;
+
+import com.google.gson.stream.JsonReader;
+import uk.ac.ebi.atlas.model.resource.AtlasResource;
+import uk.ac.ebi.atlas.model.resource.JsonFile;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+
+@Named
+public class AtlasInformationDao {
+
+    private final AtlasResource<JsonReader> atlasInformationJsonFile;
+
+    @Inject
+    public AtlasInformationDao(Path atlasInformationFilePath) {
+        atlasInformationJsonFile =
+                new JsonFile.ReadOnly(
+                        atlasInformationFilePath.getParent().toString(),
+                        atlasInformationFilePath.getFileName().toString());
+    }
+
+    public ArrayList<String> fetchAll() throws IOException {
+        ArrayList<String> atlasInformation = new ArrayList<>();
+        try (JsonReader reader = atlasInformationJsonFile.get()) {
+            reader.beginObject();
+            while (reader.hasNext()) {
+                String name = reader.nextName();
+                switch(name){
+                    case "ensembl": atlasInformation.add("Ensembl: " + reader.nextString()); break;
```

Comment:
It is not the DAO's responsibility to format the data, its only responsibility is to retrieve it. The UI is responsible for saying what this data means. The side-effect of this implementation is that you end up doing strange things in the `.jsp` to add commas. What would you do if you wanted to display the EFO version, for example, on a different page altogether? You'd have to parse each string in the list and find out which one is the EFO version, then extract the number.

Can you think of a better data structure that would allow you to retrieve the value for any given piece of metadata? Also, can you think of a way to identify the data without having to pass strings around? Hint: read about [enums in Java](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)

Reply:
Thank you for your comments. I have changed the data structure into Hashmap to retrieve any piece of metadata. However, I did not see the point of `enums` usage here, since the data we read from the file is the type of `String`, why we do not just pass strings around directly?

Reply:
What happens if we change the structure of the file? Let's say we want to rename "paraSite" to "Fred". Let's also imagine that you used the string "paraSite" to extract the information from the map in 3 different classes. You'll end up having to change 3 different files to change the property name. However, if you use enums, you only need to change 1 file, i.e. the definition of the enum. An enum would also allow you to make use of more advanced features of your IDE to find out where they're used. If you use strings, your only option is using the Find function to search across all the files in the project. 

You're probably thinking, "well I know exactly which files need changing, I won't need to use Find anyway". But what if 6 months have gone by? What if it's not you changing the files, but another developer?

Reply:
Wow, I see. I like your "probably thinking" paragraph. :) I will have a corresponding change. Thank you for your detailed description.